lib/spack/llnl/util/filesystem.py

```diff
@@ -706,7 +729,11 @@ def macro_definitions(self):
         >>> h = HeaderList(['/dir1/a.h', '/dir1/b.h', '/dir2/c.h'])
         >>> h.add_macro('-DBOOST_LIB_NAME=boost_regex')
         >>> h.add_macro('-DBOOST_DYN_LINK')
-        >>> assert h.macro_definitions == '-DBOOST_LIB_NAME=boost_regex -DBOOST_DYN_LINK'  # noqa
```

Comment:
Fun fact: comments inside of triple quotes aren't actually comments, even though `vim` seems to think so. "# noqa" was showing up in the docs.